src/sidebar/components/selection-tabs.js

```diff
@@ -1,49 +1,174 @@
+const { withServices } = require('../util/service-context');
+
+/**
+ *  Tabbed display of annotations and notes.
+ */
+
+function SelectionTabs({
+  isWaitingToAnchorAnnotations,
+  isLoading,
+  selectedTab,
+  totalAnnotations,
+  totalNotes,
+  totalOrphans,
+  settings,
+  session,
+}) {
+  // actions
+  const store = useStore(store => ({
+    clearSelectedAnnotations: store.clearSelectedAnnotations,
+    selectTab: store.selectTab,
+  }));
+
+  const isThemeClean = settings.theme === 'clean';
+
+  const selectTab = function(type) {
+    store.clearSelectedAnnotations();
+    store.selectTab(type);
+  };
+
+  const showAnnotationsUnavailableMessage = function() {
+    return (
+      selectedTab === uiConstants.TAB_ANNOTATIONS &&
+      totalAnnotations === 0 &&
+      !isWaitingToAnchorAnnotations
+    );
+  };
 
-module.exports = {
-  controllerAs: 'vm',
-  //@ngInject
-  controller: function($element, store, features, session, settings) {
-    this.TAB_ANNOTATIONS = uiConstants.TAB_ANNOTATIONS;
-    this.TAB_NOTES = uiConstants.TAB_NOTES;
-    this.TAB_ORPHANS = uiConstants.TAB_ORPHANS;
-
-    this.isThemeClean = settings.theme === 'clean';
-
-    this.enableExperimentalNewNoteButton =
-      settings.enableExperimentalNewNoteButton;
-
-    this.selectTab = function(type) {
-      store.clearSelectedAnnotations();
-      store.selectTab(type);
-    };
-
-    this.showAnnotationsUnavailableMessage = function() {
-      return (
-        this.selectedTab === this.TAB_ANNOTATIONS &&
-        this.totalAnnotations === 0 &&
-        !this.isWaitingToAnchorAnnotations
-      );
-    };
-
-    this.showNotesUnavailableMessage = function() {
-      return this.selectedTab === this.TAB_NOTES && this.totalNotes === 0;
-    };
-
-    this.showSidebarTutorial = function() {
-      return sessionUtil.shouldShowSidebarTutorial(session.state);
-    };
-  },
-  bindings: {
-    isLoading: '<',
-    isWaitingToAnchorAnnotations: '<',
-    selectedTab: '<',
-    totalAnnotations: '<',
-    totalNotes: '<',
-    totalOrphans: '<',
-  },
-  template: require('../templates/selection-tabs.html'),
+  const showNotesUnavailableMessage = function() {
+    return selectedTab === uiConstants.TAB_NOTES && totalNotes === 0;
+  };
+
+  const showSidebarTutorial = function() {
+    return sessionUtil.shouldShowSidebarTutorial(session.state);
+  };
+
+  return (
+    <div>
+      <div
+        className={classnames('selection-tabs', {
+          'selection-tabs--theme-clean': isThemeClean,
+        })}
+      >
+        <a
+          className={classnames('selection-tabs__type', {
+            'is-selected': selectedTab === uiConstants.TAB_ANNOTATIONS,
+          })}
+          href="#"
+          onClick={selectTab.bind(this, uiConstants.TAB_ANNOTATIONS)}
+          onTouchStart={selectTab.bind(this, uiConstants.TAB_ANNOTATIONS)}
+        >
+          Annotations
+          {totalAnnotations > 0 && !isWaitingToAnchorAnnotations && (
```

Comment:
Might take a look to see if there is a slightly more scannable way to present this logic.

Reply:
Might be best to fold t his into a refactor effort
Best I can see it making a Tab component but since it does not have really any special logic other than passing down props, may not be super helpful.